src/humanPose/MuriScore.js

```diff
@@ -0,0 +1,818 @@
+import {ErgonomicsData, clamp} from "./ErgonomicsData.js";
+import {JOINT_CONNECTIONS, JOINTS, getBoneName, ERGO_ANGLES, ERGO_OFFSETS} from './constants.js';
+
+/** Default configuration of thresholds for MURI calculation and visualisation. */
+const MURI_CONFIG_DEFAULT = Object.freeze({
+    trunkFrontBendAngleThresholds: [-5, 20, 45],  // degrees ; normally [0, 20, 45] but we give margin of -5 deg to accommodate imperfect accuracy of pose when upright 
+    trunkSideBendAngleThresholds: [15, 30],  // degrees
+    trunkTwistAngleThresholds: [15, 45],  // degrees
+    headFrontBendAngleThresholds: [-3, 15, 30],  // degrees ; normally [0, 15, 30] but we give margin of -3 deg to accommodate imperfect accuracy of pose when upright 
+    headSideBendAngleThresholds: [15, 30],  // degrees
+    headTwistAngleThresholds: [15, 30],  // degrees
+    footHeightDifferenceThresholds: [150], // mm   
+    lowerLegBendAngleThresholds: [60],  // degrees
+    upperArmFrontRaiseAngleThresholds: [60, 90], // degrees
+    upperArmSideRaiseAngleThresholds: [60, 90], // degrees
+    lowerArmBendAngleThresholds: [120], // degrees
+    lowerArmTwistAngleThresholds: [120],  // degrees
+    handFrontBendAngleThresholds: [-20, 30], // degrees
+    handReachDistanceThresholds: [300, 500],  // mm   //TODO: tune based on customer feedback
+
+    // Weigths for different levels of strain on a body part which are used to calculate actual score.
+    // Note: Weights are integer numbers in the latest iteration of MURI method
+    scoreWeights: [0, 1, 2, 5], 
+
+    // Definitions of score ranges for different levels of strain for each body part.
+    // This is used to assign colours for visualising MURI scores.
+    // Minimum score is always 0 and maximum score is (<the last entry in the threshold list> - 1)
+    // Example with 4 levels of strain: 0 (low), 1 (medium), 2 (high), 3-5 (extreme)
+    //    scoreLevels: [1, 2, 3, 6]
+    // Warning: Actual score values (in case of integer scoreWeights) are smaller by -1 than the thresholds   
+    trunkScoreLevels: [1, 2, 3],
+    headScoreLevels: [1, 2, 3],
+    legScoreLevels: [1, 6],  
+    upperArmScoreLevels: [1, 2, 3],
+    lowerArmScoreLevels: [1, 2, 3],
+    handScoreLevels: [1, 2, 3],
+    overallScoreLevels: [0 /* MIN_MURI_SCORE */, 30 /* MAX_MURI_SCORE + 1 */]  // TODO: tune based on customer feedback
+});
+
+/** Modifiable configuration of thresholds for MURI calculation. */
+export const MURI_CONFIG = Object.assign({}, MURI_CONFIG_DEFAULT);
+
+const MURI_COLOR_START = MotionStudyColors.fade(MotionStudyColors.green);
+const MURI_COLOR_END = MotionStudyColors.fade(MotionStudyColors.red);
+
+export const MIN_MURI_SCORE = 0
+// Note: calculate precisely based on final calculation of overall muri score.
+export const MAX_MURI_SCORE = 6 * MURI_CONFIG_DEFAULT.scoreWeights[2] /* trunk + head */ + 2 * 6 * MURI_CONFIG_DEFAULT.scoreWeights[2] /* arms */ + 1 * MURI_CONFIG_DEFAULT.scoreWeights[3] /* legs */ // == 41 
```

Comment:
Should the first one here be `scoreWeights[1]`?

Reply:
I think this could be better explained as "MAX_MURI_SCORE_TRUNK_HEAD = 6 joints * max score for those 6 joints is scoreWeights[2]" and so on and so forth. Refactoring out to individual variables for each part might help, also explaining the source of each magic number